common/tests/real_thandle_helper_ut/real_thandle_helper_ut.c

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "macro_utils/macro_utils.h" // IWYU pragma: keep
+
+#include "umock_c/umock_c.h"
+
+#include "testrunnerswitcher.h"
+
+#define ENABLE_MOCKS
+#include "c_pal/gballoc_hl.h"
+#include "c_pal/gballoc_hl_redirect.h" // IWYU pragma: keep
+#include "c_pal/interlocked.h" // IWYU pragma: keep
+#include "c_pal/interlocked_hl.h" // IWYU pragma: keep
+#include "c_pal/thandle.h"
+#include "c_pal/thandle_ll.h"
+
+struct MOCKED_STRUCT_TAG; // IWYU pragma: private
+typedef struct MOCKED_STRUCT_TAG MOCKED_STRUCT; // IWYU pragma: private
+THANDLE_TYPE_DECLARE(MOCKED_STRUCT);
+
+#undef ENABLE_MOCKS
+
+#include "umock_c/umock_c_prod.h"
+
+#include "real_gballoc_hl.h"
+#include "real_interlocked.h" // IWYU pragma: keep
+#include "real_interlocked_hl.h" // IWYU pragma: keep
+
+#include "real_thandle_helper.h"
+
+#include "real_interlocked_renames.h" // IWYU pragma: keep
+
+typedef struct MOCKED_STRUCT_TAG
+{
+    uint8_t dummy;
+} MOCKED_STRUCT;
+
+REAL_THANDLE_DECLARE(MOCKED_STRUCT);
+
+REAL_THANDLE_DEFINE(MOCKED_STRUCT);
+
+#include "real_interlocked_undo_rename.h" // IWYU pragma: keep
```

Comment:
>real_interlocked_undo_rename [](http://example.com/codeflow?start=10&length=28)

consider a test that checks the normal `umock_c_get_expected_calls` mock things to make sure the interlocked functions generated by THANDLE are ignored.

I guess you could get even fancier and actually call an interlocked function (so you expect the one you call manually to be recorded by the mocking but the calls internal to THANDLE are not).